ibm_spectrum_lsf/datadog_checks/ibm_spectrum_lsf/processors.py

```diff
@@ -539,3 +547,47 @@ def process_metrics(self) -> list[LSFMetric]:
                 metrics.append(LSFMetric(f"{self.prefix}.{metric_name}.{aggr}", metric_value, self.base_tags))
 
         return metrics
+
+
+class BHistProcessor(LSFMetricsProcessor):
+    def __init__(self, client: LSFClient, config: InstanceConfig, logger: CheckLoggingAdapter, base_tags: list[str]):
+        super().__init__(
+            name='bhist',
+            prefix='job.completed',
+            expected_columns=10,
+            delimiter=None,
+            client=client,
+            config=config,
+            logger=logger,
+            base_tags=base_tags,
+        )
+        self.last_check_time = get_current_datetime().strftime('%Y/%m/%d/%H:%M')
+
+    def run_lsf_command(self) -> tuple[str, str, int]:
+        start_time = self.last_check_time
+        end_time = get_current_datetime().strftime('%Y/%m/%d/%H:%M')
+        self.log.trace("Last check time: %s, end time: %s", start_time, end_time)
+        if start_time == end_time:
+            self.log.trace("Start time %s is equal to end time %s, going back 1 minute", start_time, end_time)
+            # the highest granularity is 1 minute, so we need to go back 1 minute if collection interval < 60
+            start_time = (get_current_datetime() - timedelta(minutes=1)).strftime('%Y/%m/%d/%H:%M')
+        self.last_check_time = end_time
+        return self.client.bhist(start_time, end_time)
+
+    def process_metrics(self) -> list[LSFMetric]:
+        tags = [
+            LSFTagMapping('job_id', 0, transform_job_with_task),
+            LSFTagMapping('task_id', 0, transform_task_id),
+            LSFTagMapping('user', 1, transform_tag),
+            LSFTagMapping('job_name', 2, transform_job_name),
+        ]
+        metrics = [
+            LSFMetricMapping('pending', 3, transform_float),
+            LSFMetricMapping('pending_user_suspended', 4, transform_float),
+            LSFMetricMapping('running', 5, transform_float),
```

Comment:
Given that this command has a start and end time, how does it handle things that transitioned during this time frame? For example let's assume that a job was pending in 1:01:00 and then it transitioned to running 1:01:30 and you're querying from 1:02:00 - 1:01:00. How will it show up?

Reply:
How long the job spent in the pending state and how long it spent in the running state.

Reply:
Ahh okay. That makes sense then!